javascript/680-Valid-Palindrome-II.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Linear Time.
+ * Time O(N) | Space O(1)
+ * https://leetcode.com/problems/valid-palindrome-ii
+ * @param {string} s
+ * @return {boolean}
+ */
+//
+var validPalindrome = function(s) {
+
+    let left = 0;
+    let right = s.length - 1;
+
+    while (left < right) {
+        if (s[left] !== s[right]) {
+
+            if (
```

Comment:
This can be cleaned up a bit more

```js
const isValid = isPalindrome(left, right) || isPalindrome(left + 1, right + 1)
if (isValid) return true;


const isPalindrome = (s, left, right) => {
	const s = s.slice(left, right)
	return s === reOrder(s);
}

const reOrder = (s) => s
	.split('')
	.reverse()
	.join('')
```

Reply:
Hey @aakhtar3,  We can't reassign s in isPalindrome as this is already coming as a parameter. You forgot to pass the first argument in isPalindrome on the very first line. I did the suggested changes. Please check now.

Reply:
Oh, sorry I meant you can't re-create the variable by the same name in the isPalindrome.